lib/login.rb

```diff
@@ -0,0 +1,33 @@
+require "pg"
+require "bcrypt"
+require_relative "db_connection"
+
+class Login
+  attr_reader :username, :name, :email
+
+  def initialize(username:, name:, email:)
+    @username = username
+    @name = name
+    @email = email
+  end
+
+  def self.authenticate(email:, password:)
+    user = DBConnection.query("SELECT * FROM users WHERE email = '#{email}'")
+    return unless user.any? #is the user registered?
+    return unless BCrypt::Password.new(user[0]["password"]) == password
```

Comment:
it's nice that you have used the encrypted password!

Reply:
Better to keep it secure!😇